File: MiniFTPServer/vb/LibFTPServ.ConvertedToVBNet/VFS/VFSVirtualItem.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace VFS
	''' <summary>
	''' Base class of VFS virtual items
	''' </summary>
	Public MustInherit Class VFSVirtualItem
		Private _name As String
		Private _size As Long
		Private _date As DateTime

		''' <summary>
		''' The full path of the item in the VFS. 
		''' </summary>
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set
				_name = value
			End Set
		End Property

		''' <summary>
		''' The size of the item in the VFS
		''' </summary>
		Public Property Size() As Long
			Get
				Return _size
			End Get
			Set
				_size = value
			End Set
		End Property

		''' <summary>
		''' Item date in the VFS
		''' </summary>
		Public Property [Date]() As DateTime
			Get
				Return _date
			End Get
			Set
				_date = value
			End Set
		End Property

		''' <summary>
		''' Creates a new instance of VFSVirtualItem
		''' </summary>
		''' <param name="name">The full path of the item in the VFS. </param>
		Public Sub New(name As String)
			_name = name
			_date = DateTime.Now
			_size = 0
		End Sub

		''' <summary>
		''' Creates a new instance of VFSVirtualItem
		''' </summary>
		''' <param name="name">The full path of the item in the VFS. </param>
		''' <param name="date">Item date in the VFS</param>
		''' <param name="size">The size of the item in the VFS</param>
		Public Sub New(name As String, [date] As DateTime, size As Long)
			_name = name
			_date = [date]
			_size = size
		End Sub
	End Class
End Namespace

File: MiniFTPServer/LibFTPServ/VFS/VFSVirtualItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace LibFTPServ.VFS
{
    /// <summary>
    /// Base class of VFS virtual items
    /// </summary>
    public abstract class VFSVirtualItem
    {
        private string _name;
        private long _size;
        private DateTime _date;

        /// <summary>
        /// The full path of the item in the VFS. 
        /// </summary>
        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        /// <summary>
        /// The size of the item in the VFS
        /// </summary>
        public long Size
        {
            get { return _size; }
            set { _size = value; }
        }

        /// <summary>
        /// Item date in the VFS
        /// </summary>
        public DateTime Date
        {
            get { return _date; }
            set { _date = value; }
        }

        /// <summary>
        /// Creates a new instance of VFSVirtualItem
        /// </summary>
        /// <param name="name">The full path of the item in the VFS. </param>
        public VFSVirtualItem(string name)
        {
            _name = name;
            _date = DateTime.Now;
            _size = 0;
        }

        /// <summary>
        /// Creates a new instance of VFSVirtualItem
        /// </summary>
        /// <param name="name">The full path of the item in the VFS. </param>
        /// <param name="date">Item date in the VFS</param>
        /// <param name="size">The size of the item in the VFS</param>
        public VFSVirtualItem(string name, DateTime date, long size)
        {
            _name = name;
            _date = date;
            _size = size;
        }
    }
}
